build-android/app/src/main/java/com/google/firebase/codelab/friendlychat/ButtonObserver.kt

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.firebase.codelab.friendlychat
+
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.Button
+
+class ButtonObserver(private val mButton: Button) : TextWatcher {
+    override fun onTextChanged(charSequence: CharSequence, start: Int, count: Int, after: Int) {
+        mButton.isEnabled = charSequence.toString().trim { it < ' ' }.isNotEmpty()
```

Comment:
I might have missed this on the last PR but when going to Kotlin we should stop using the `mFoo` style since that's Java style. Just `button` works.

Reply:
This comment applies to all files.

Reply:
👍  thanks for the heads up. I'll fix across all files